permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -40,3 +40,1002 @@ impl IsGrantAllowed for ProhibitGrant {
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&carol_id, &find_bronze, &wsv)
+                    .is_err());
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_bronze, &wsv)
+                    .is_err());
+            }
+        }
+
+        #[test]
+        fn find_assets_by_account_id() {
+            let TestEnv {
+                alice_id,
+                bob_id,
+                carol_id,
+                wsv,
+                ..
+            } = TestEnv::new();
+
+            let op = QueryBox::FindAssetsByAccountId(FindAssetsByAccountId::new(alice_id.clone()));
+
+            {
+                let only_accounts_domain: IsQueryAllowedBoxed = query::OnlyAccountsDomain.into();
+
+                assert!(only_accounts_domain.check(&alice_id, &op, &wsv).is_ok());
+                assert!(only_accounts_domain.check(&bob_id, &op, &wsv).is_ok());
+                assert!(only_accounts_domain.check(&carol_id, &op, &wsv).is_err());
+            }
+
+            {
+                let only_accounts_data: IsQueryAllowedBoxed = query::OnlyAccountsData.into();
+
+                assert!(only_accounts_data.check(&alice_id, &op, &wsv).is_ok());
+                assert!(only_accounts_data.check(&bob_id, &op, &wsv).is_err());
+                assert!(only_accounts_data.check(&carol_id, &op, &wsv).is_err());
+            }
+        }
+
+        #[test]
+        fn find_assets_by_asset_definition_id() {
+            let TestEnv {
+                alice_id,
+                bob_id,
+                carol_id,
+                gold_asset_definition_id,
+                wsv,
+                ..
+            } = TestEnv::new();
+
+            let find_gold = QueryBox::FindAssetsByAssetDefinitionId(
+                FindAssetsByAssetDefinitionId::new(gold_asset_definition_id),
+            );
+
+            {
+                let only_accounts_domain: IsQueryAllowedBoxed = query::OnlyAccountsDomain.into();
+
+                assert!(only_accounts_domain
+                    .check(&alice_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_domain
+                    .check(&bob_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_domain
+                    .check(&carol_id, &find_gold, &wsv)
+                    .is_err());
+            }
+
+            {
+                let only_accounts_data: IsQueryAllowedBoxed = query::OnlyAccountsData.into();
+
+                assert!(only_accounts_data
+                    .check(&alice_id, &find_gold, &wsv)
+                    .is_err());
+                assert!(only_accounts_data.check(&bob_id, &find_gold, &wsv).is_err());
+                assert!(only_accounts_data
+                    .check(&carol_id, &find_gold, &wsv)
+                    .is_err());
+            }
+        }
+    }
+
+    #[test]
+    fn find_assets_by_domain_id() {
```

Comment:
Is this supposed to be outside `mod queries`?

Reply:
If so, I'd add an explicit `mod` that explains what these tests are.

Reply:
Ah, that's after the wrong rebase. Thanks, fixed.